### PR TITLE
Chore: adds migration for newVariables

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Relates #22564

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/dashboard/state/DashboardMigrator.ts [hunk 1]
```diff
@@ -497,6 +497,13 @@ export class DashboardMigrator {
       });
     }
 
+    if (oldVersion < 23) {
+      if (getConfig().featureToggles.newVariables) {
+        this.dashboard.variables.list = cloneDeep(this.dashboard.templating.list);
+        this.dashboard.templating.list = [];
```
Comment (hugohaggmark) [inline public/app/features/dashboard/state/DashboardMigrator.ts]: Not sure about setting `this.dashboard.templating.list` to empty might be hard to debug errors. WDYT @mckn @torkelo.

Comment (dprokop) [reply public/app/features/dashboard/state/DashboardMigrator.ts]: @hugohaggmark I think we should just delete it

**Review by torkelo — COMMENTED** [review]
Looks good but my only question is do we need this migration? Can't we just make the new variable system use templating.list instead? A big migration like this will cause a big diff in dashboard json so if we can avoid it would be preferable. 

It does not look like there are any model changes (and if there where they could be done on templating.list)

File: public/app/features/dashboard/state/DashboardMigrator.ts (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/state/DashboardMigrator.ts]: @dprokop reallly? Maybe we should do this at the same time we remove the featureToggle is my thinking?

Comment (dprokop) [reply public/app/features/dashboard/state/DashboardMigrator.ts]: @hugohaggmark that's what I meant, sorry for not being clear

**Comment by hugohaggmark** [comment]
> Looks good but my only question is do we need this migration? Can't we just make the new variable system use templating.list instead? A big migration like this will cause a big diff in dashboard json so if we can avoid it would be preferable.
> 
> It does not look like there are any model changes (and if there where they could be done on templating.list)

Mind blown! That is probably a better approach. Might be tricky though. I'll look into that in a separate PR.

**Comment by mckn** [comment]
> Looks good but my only question is do we need this migration? Can't we just make the new variable system use templating.list instead? A big migration like this will cause a big diff in dashboard json so if we can avoid it would be preferable.
> 
> It does not look like there are any model changes (and if there where they could be done on templating.list)

We talked about that when we started but decided to start with a separate variable. But I totally agree with this. Hopefully we have not changed any contracts in the data so it should work with just switching.

**Comment by hugohaggmark** [comment]
Closing this in favor of #23107.

**Pull request closed without merging**